test/extended/default_forcePull_test.go

```diff
@@ -0,0 +1,234 @@
+while this test is running and compare results.  Restarting your docker daemon, assuming you can ping docker.io quickly, could
+be a quick fix.
+*/
+
+func corruptImage(corruptee, corruptor, strategy string) {
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Calling docker tag to corrupt %s builder image %s by tagging %s", time.Now().Format(time.RFC850), strategy, corruptee, corruptor))
+
+	cerr := exutil.CorruptImage(corruptee, corruptor)
+
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Tagging %s to %s complete with err %v", time.Now().Format(time.RFC850), corruptor, corruptee, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+}
+
+func resetImage() {
+	// reset both images just to be sure
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Calling docker pull to reset images", time.Now().Format(time.RFC850)))
+
+	serr := exutil.ResetImage(s2iDockBldr)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset pull for %s complete with err %v", time.Now().Format(time.RFC850), s2iDockBldr, serr))
+	Expect(serr).NotTo(HaveOccurred())
+
+	cerr := exutil.ResetImage(custBldr)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset pull for %s complete with err %v", time.Now().Format(time.RFC850), custBldr, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+
+	// a small sleep between a potential "AfterEach" and the next potential "BeforeEach", just to make sure the docker daemon has processed our reset of images
+	time.Sleep(1 * time.Second)
+}
+
+func verifyImagesSame(strategy string) {
+	s2iBldrLatestID := ""
+	custBldrLatestID := ""
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	cerr := exutil.CompareImages(s2iComp, custComp, &s2iBldrLatestID, &custBldrLatestID)
+
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s false ******:  compare image - %s, %s, %s, %s, %v", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, s2iBldrLatestID, custBldrLatestID, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+	Expect(len(s2iBldrLatestID) > 0).Should(BeTrue())
+	Expect(len(custBldrLatestID) > 0).Should(BeTrue())
+	Expect(s2iBldrLatestID).To(Equal(custBldrLatestID))
+}
+
+func verifyImagesDifferent(strategy string) {
+	s2iBldrLatestID := ""
+	custBldrLatestID := ""
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	cerr := exutil.CompareImages(s2iComp, custComp, &s2iBldrLatestID, &custBldrLatestID)
+
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s true ******:  compare image - %s, %s, %s, %s, %v", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, s2iBldrLatestID, custBldrLatestID, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+	Expect(len(s2iBldrLatestID) > 0).Should(BeTrue())
+	Expect(len(custBldrLatestID) > 0).Should(BeTrue())
+	Expect(s2iBldrLatestID != custBldrLatestID).Should(BeTrue())
+}
+
+func sleepDuringBuild(context string) {
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s ******:   sleep 10 seconds, allow image to be pulled", time.Now().Format(time.RFC850), context))
+	time.Sleep(10 * time.Second)
```

Comment:
again can we be more proactive here about checking whatever it is we're waiting for?